From 52e9263dfa3c04f8f8e99df03cbc51cc5c1ceb45 Mon Sep 17 00:00:00 2001
From: Nirbheek Chauhan <nirbheek@centricular.com>
Date: Sun, 17 Apr 2016 12:39:12 +0530
Subject: [PATCH] giomodule: do not try to load modules from gio-module-dir in
 IOS

There is no gio-module-dir for ios. When it tries to load on a real
device it gets ignored because the path doesn't exist. When it tries
on the simulator, it will print assertions because the path might
exist in the user machine, but it will fail to load the modules

Original patch by thiago.sousa.santos@collabora.com; blindly ported to 2.48 by
Nirbheek
---
 gio/giomodule.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gio/giomodule.c b/gio/giomodule.c
index b97aa78..d39b74f 100644
--- a/gio/giomodule.c
+++ b/gio/giomodule.c
@@ -1081,7 +1081,9 @@ _g_io_modules_ensure_loaded (void)
       /* Then load the compiled in path */
       module_dir = get_gio_module_dir ();
 
+#ifndef GST_SDK_IOS
       g_io_modules_scan_all_in_directory_with_scope (module_dir, scope);
+#endif
       g_free (module_dir);
 
 #ifdef GST_SDK_GLIB_GIO_DISTRO_GIO_MODULE_PATH
-- 
2.7.2.windows.1

